java/client/src/org/openqa/selenium/support/events/EventFiringWebDriver.java

```diff
@@ -63,7 +64,7 @@
  * {@link WebDriverEventListener}, e&#46;g&#46; for logging purposes.
  */
 public class EventFiringWebDriver implements WebDriver, JavascriptExecutor, TakesScreenshot,
-    WrapsDriver, HasInputDevices, HasTouchScreen {
+    WrapsDriver, HasInputDevices, HasTouchScreen, Alert {
```

Comment:
your Mock of WebDriver needs to respond to driver.switchTo() and return a mock of a TargetLocator and then the targetLocator.alert() needs to return your mocked Alert.